### PR TITLE
Smt lib migration investigation tests

### PR DESCRIPTION
**Pull request by Kolezhniuk**

### CODE REVIEW

File: contracts/mocks/SmtV2Mock.sol [hunk 1]
```diff
@@ -1,183 +1,315 @@
...
+        public
+        view
+        returns (uint256)
+    {
+        return self.rootHistory.length;
     }
 
-    function getRootHistory(uint256 startIndex, uint256 endIndex)
+    /**
+     * =============== ADEED FUNCTION ===============
```
Comment (AndriianChestnykh) [inline contracts/mocks/SmtV2Mock.sol:115]: ADDED

File: contracts/mocks/SmtV2Mock.sol [hunk 2]
```diff
@@ -1,183 +1,315 @@
...
+            self
+                .rootTransitions[self.rootHistory.length - 2]
+                .replacedAtTimestamp = _timestamp;
+            self
+                .rootTransitions[self.rootHistory.length - 2]
+                .replacedAtBlock = _blockNumber;
+            self.rootTransitions[self.rootHistory.length - 2].replacedBy = self
+                .root;
+        }
+        _replacedBy = 123;
```
Comment (AndriianChestnykh) [inline contracts/mocks/SmtV2Mock.sol:235]: What is the logic for this assignment?

File: test/smt/smt-upgrade.test.ts [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    //deploy new smt library version
+    const smtV2Lib = await stateDeployHelper.deploySmt(
+      poseidon2Elements.address,
+      poseidon3Elements.address,
+      "SmtV3Mock"
+    );
+
+    // upgrade smt library version
+    const StateV3MockFactory = await ethers.getContractFactory("StateV3Mock", {
+      libraries: { SmtV3Mock: smtV2Lib.address },
```
Comment (AndriianChestnykh) [inline test/smt/smt-upgrade.test.ts:127]: It's hard mentally to follow when mixing different versions like `const smtV2Lib` and `SmtV3Mock`

File: contracts/mocks/SmtV2Mock.sol (hunk 2 above)
Comment (Kolezhniuk) [reply contracts/mocks/SmtV2Mock.sol:235]: The main goal here is to verify upgrading signature of function (added additional _replacedBy parameter)

File: test/smt/smt-upgrade.test.ts (hunk 3 above)
Comment (Kolezhniuk) [reply test/smt/smt-upgrade.test.ts]: fixed

File: contracts/mocks/SmtV2Mock.sol (hunk 1 above)
Comment (Kolezhniuk) [reply contracts/mocks/SmtV2Mock.sol]: fixed

**Pull request merged**